packages/swap/src/routes/quote.ts

```diff
@@ -21,14 +22,68 @@ import { resultify } from '../utils/promise';
 import {
   getMinimumEgressAmount,
   getEgressFee,
-  getIngressFee,
   validateSwapAmount,
+  getBoostPoolsDepth,
 } from '../utils/rpc';
 import ServiceError from '../utils/ServiceError';
 import { estimateSwapDuration } from '../utils/swap';
 
 const getPoolQuoteResult = resultify(getPoolQuote);
 
+export const getBoostedPoolQuoteResult = async (query: ParsedQuoteParams) => {
+  const { srcAsset, destAsset, brokerCommissionBps, amount } = query;
+  const fees: SwapFee[] = [];
+
+  const assetBoostPoolsDepth = await getBoostPoolsDepth({ asset: srcAsset });
+
+  const effectiveBoostFeeBps = await getBoostFeeBpsForAmount({
+    amount: BigInt(amount),
+    assetBoostPoolsDepth,
+  });
+
+  if (effectiveBoostFeeBps === undefined) return undefined;
+
+  const boostFee = getPipAmountFromAmount(amount, effectiveBoostFeeBps);
+  fees.push(buildFee(srcAsset, 'BOOST', boostFee));
+
+  if (amount <= 0n) {
```

Comment:
amount wasn't modified, is it supposed to be?

Reply:
I missed that, there should be a check for this. Thanks for pointing out.